contracts/actions/common/SetApproval.sol

```diff
@@ -38,7 +39,12 @@ contract SetApproval is Executable, UseStore {
 
     IERC20(approval.asset).safeApprove(approval.delegate, actualApprovalAmount);
 
-    emit Action(SET_APPROVAL_ACTION, bytes(abi.encode(actualApprovalAmount)));
+    IEventEmitter eventEmitter = IEventEmitter(registry.getRegisteredService(EVENT_EMITTER));
+    eventEmitter.emitActionEvent(
+      SET_APPROVAL_ACTION,
+      msg.sender,
```

Comment:
Why `msg.sender` hre and not `address(this)` ?

Reply:
Mistake